File: 896.monotonic-array/solution.scala
object Solution {
    def isMonotonic(A: Array[Int]): Boolean = {
        var inc = 0
        var dec = 0
        for (i <- 1 until A.length) {
            val d = A(i) - A(i - 1)
            if (d >= 0) inc += 1
            if (d <= 0) dec += 1
        }
        Math.max(inc, dec) >= A.length - 1
    }
}

File: 896.monotonic-array/solution.go
func isMonotonic(nums []int) bool {
    lt, gt := 0, 0
    for i := 1; i < len(nums); i++ {
        if nums[i - 1] < nums[i] {
            lt++
        }
        if nums[i - 1] > nums[i] {
            gt++
        }
    }
    return lt == 0 || gt == 0 
}